### DISCUSSION TITLE
How does merging work?

### DISCUSSION THREAD

**Discussion by emrecio** [keepassxreboot/keepassxc#11561] [Q&A]
So I want an overview on how the merging exactly works so I know what is the appropriate order... if applicable to merge two or more files together.

Simple scenario: So I have two files a1.kdbx and b1.kdbx.

Both files have x1, and y1 in their database, all other entries are the same. The two files diverged (became out of synch) after 2024-11-30 0000h.

No editing on y1 has been done in a1 (initial revision)
I edit entry x1 in a1 on 2024-12-02 0300h.
I edit entry z1 in a1 on 2024-12-02 0300h.

I edit entry x1 in b1 on 2024-12-03 0300h.
I edit entry y1 in b1 on 2024-12-03 0300h.
I edit entry z1 in b1 on 2024-12-01 0300h.

a1 is my primary file that I try to keep in synch via git with other machines.

If I open a1 and merge b1 into a1, what happens? 
- x1 will be overwritten in a1 by b1's entry because b1 is more recent?
- y1 will be updated in a1 by b1 entry because b1 is more recent?
- z1 will be left alone in a1 because it's an older update in b1 for z1?

**Comment by droidmonkey** [comment]
The only factor used is modified date and uuid. If uuid matches, the latest modified date entry is used as current and the other is pushed into history.
